### PR TITLE
Replace KMS with Secret Manager

### PR DESCRIPTION
**Pull request by peterschen**
This PR replaces the usage of KMS with Secret Manager.

### CODE REVIEW

File: ad-joining/register-computer/main.py [hunk 1]
```diff
@@ -68,10 +68,20 @@ def __read_ad_password():
...
-        # Decrypt password cipher using the Cloud KMS key provided.
-        return google.cloud.kms_v1.KeyManagementServiceClient().decrypt(
-            name=__read_required_setting("CLOUDKMS_KEY"),
-            ciphertext=base64.b64decode(__read_required_setting("AD_PASSWORD_CIPHER"))).plaintext.decode("utf-8").strip()
+        client = secretmanager.SecretManagerServiceClient()
+        
+        # If the Service Account does not have permissions
+        # to access the secret an Exception will be raised
+        try:        
+            name = client.secret_version_path(__read_required_setting("SECRET_PROJECT_ID"),
```
Comment (jpassing) [inline ad-joining/register-computer/main.py:76]: Inconsistent parameter indentation

File: ad-joining/register-computer/main.py [hunk 2]
```diff
@@ -68,10 +68,20 @@ def __read_ad_password():
...
+        
+        # If the Service Account does not have permissions
+        # to access the secret an Exception will be raised
+        try:        
+            name = client.secret_version_path(__read_required_setting("SECRET_PROJECT_ID"), 
+                __read_required_setting("SECRET_NAME"), __read_required_setting("SECRET_VERSION"))
+            response = client.access_secret_version(name)
+            return response.payload.data.decode("UTF-8")
+        except Exception as e:
+            logging.exception("Could not retrieve secret from Secret Manager: %s" % e)
```
Comment (jpassing) [inline ad-joining/register-computer/main.py:81]: Add a `throw` with a more specific error message than the one below

**Review by jpassing — APPROVED** [review]
just 2 nits

File: ad-joining/register-computer/main.py (hunk 1 above)
Comment (peterschen) [reply ad-joining/register-computer/main.py]: Changed.

File: ad-joining/register-computer/main.py [hunk 3]
```diff
@@ -68,10 +68,21 @@ def __read_ad_password():
...
+        client = secretmanager.SecretManagerServiceClient()
+        
+        # If the Service Account does not have permissions
+        # to access the secret an Exception will be raised
+        try:        
+            name = client.secret_version_path(
+                    __read_required_setting("SECRET_PROJECT_ID"), 
+                    __read_required_setting("SECRET_NAME"), 
+                    __read_required_setting("SECRET_VERSION"))
+            response = client.access_secret_version(name)
```
Comment (idofl) [inline ad-joining/register-computer/main.py:80]: Running the code with **client.access_secret_version(name)** produces the following error:
_Could not retrieve secret from Secret Manager: Invalid constructor input for AccessSecretVersionRequest: 'projects/*/secrets/*/versions/*''_ (with actual values where * is).

I was able to resolve this by replacing this call with the following, as [demonstrated in public documentation](https://cloud.google.com/secret-manager/docs/creating-and-accessing-secrets#secretmanager-access-secret-version-python): 
client.access_secret_version(request={"name": name})

Comment (idofl) [reply ad-joining/register-computer/main.py:80]: Looking at the requirements.txt, the code was using v1.0 of the API (with requirement >=1.0.0). Looking at this:
https://googleapis.dev/python/secretmanager/latest/UPGRADING.html

It seems that v2.0 changed the way we need to send parameters to retrieve a secret.
I will push a fix for this.

**Pull request merged**